project-base/storefront/components/Pages/CategoryDetail/CategoryBestsellers/CategoryBestsellers.tsx

```diff
@@ -0,0 +1,63 @@
+import { Button } from 'components/Forms/Button/Button';
+import { ListedProductFragmentApi } from 'graphql/generated';
+import { useGtmSliderProductListViewEvent } from 'gtm/hooks/productList/useGtmSliderProductListViewEvent';
+import { GtmProductListNameType } from 'gtm/types/enums';
+import { useComponentUpdate } from 'hooks/helpers/useComponentUpdate';
+import { useTypedTranslationFunction } from 'hooks/typescript/useTypedTranslationFunction';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { CategoryBestsellersListItem } from './CategoryBestsellersListItem';
+
+const NUMBER_OF_VISIBLE_ITEMS = 3;
+
+type CategoryBestsellersProps = {
+    products: ListedProductFragmentApi[];
+};
+
+export const CategoryBestsellers: FC<CategoryBestsellersProps> = ({ products }) => {
+    const t = useTypedTranslationFunction();
+    const router = useRouter();
+    const [isCollapsed, setIsCollapsed] = useState(false);
+
+    useComponentUpdate(() => setIsCollapsed(false), [router.asPath]);
+    useGtmSliderProductListViewEvent(products, GtmProductListNameType.bestsellers);
+
+    if (products.length === 0) {
+        return null;
+    }
```

Comment:
Since you are passing data from parent doesn't make much sense to me to run logic until these lines. What about put this check to don't even show this component in case we don't have any data?

Reply:
Good point, rewritten.